src/FilterList.js

```diff
@@ -4,28 +4,48 @@ import { observer } from 'mobx-react'
 import { Dynamic, SpacedList } from './layout'
 import InjectTreeNode from './utils/injectTreeNode'
 
-export let Label = x => (
-  <b style={{ display: 'block', margin: '10px 0' }} {...x} />
-)
+export let Label = observer(x => (
+  <div style={{ margin: '10px 0' }}>
+    <b {...x} />
+    {x.tree && (
```

Comment:
instead of prop drilling tree down from FieldLabel, you can read it off the context using `inject`, similar to how inject tree node works

Reply:
done